File: EssentialToolsCs/EssentialToolsCs/Infrastructure/NinjectDependencyResolver.cs
namespace EssentialToolsCs.Infrastructure
{
    using System;
    using System.Collections.Generic;
    using System.Web.Mvc;
    using Models;
    using Ninject;
    using Ninject.Web.Common;

    public class NinjectDependencyResolver : IDependencyResolver
    {
        private readonly IKernel kernel;

        public NinjectDependencyResolver(IKernel kernelParam)
        {
            kernel = kernelParam;
            AddBindings();
        }
        public object GetService(Type serviceType)
        {
            return kernel.TryGet(serviceType);
        }

        public IEnumerable<object> GetServices(Type serviceType)
        {
            return kernel.GetAll(serviceType);
        }

        private void AddBindings()
        {
            // kernel.Bind<IValueCalculator>().To<LinqValueCalculator>();
            kernel.Bind<IValueCalculator>().To<LinqValueCalculator>().InRequestScope();

            // kernel.Bind<IDiscountHelper>().To<DefaultDiscountHelper>();
            // property injection
            // kernel.Bind<IDiscountHelper>().To<DefaultDiscountHelper>().WithPropertyValue("DiscountSize", 50M);
            // constructor injection
            kernel.Bind<IDiscountHelper>().To<DefaultDiscountHelper>().WithConstructorArgument("discountParam", 50M);
            kernel.Bind<IDiscountHelper>().To<FlexibleDiscountHelper>().WhenInjectedInto<LinqValueCalculator>();
        }
    }
}

File: EssentialToolsVb/EssentialToolsVb/Infrastructure/NinjectDependencyResolver.vb
Imports System.Web.Mvc
Imports Ninject

Public Class NinjectDependencyResolver
    Implements IDependencyResolver

    Private ReadOnly kernel As IKernel

    Public Sub New(kernelParam As IKernel)
        kernel = kernelParam
        AddBindings()
    End Sub

    Public Function GetService(serviceType As Type) As Object Implements IDependencyResolver.GetService
        Return kernel.TryGet(serviceType)
    End Function

    Public Function GetServices(serviceType As Type) As IEnumerable(Of Object) Implements IDependencyResolver.GetServices
        Return kernel.GetAll(serviceType)
    End Function

    Private Sub AddBindings()
        kernel.Bind(Of IValueCalculator).To(Of LinqValueCalculator)()
        ' kernel.Bind(Of IDiscountHelper).To(Of DefaultDiscountHelper)()
        ' property injection
        ' kernel.Bind(Of IDiscountHelper).To(Of DefaultDiscountHelper)().WithPropertyValue("DiscountSize", 50D)
        ' constructor injection
        kernel.Bind(Of IDiscountHelper).To(Of DefaultDiscountHelper)().WithConstructorArgument("discountParam", 50D)
    End Sub
End Class